src/pages/admin/event/index.tsx

```diff
@@ -34,11 +58,83 @@ const ManageAllEventsPage: NextPage<IProps> = (props: IProps) => {
     }
   }, [allEvents, futureEvents, mode, pastEvents]);
 
+  const filteredEvents = visibleEvents
+    .filter(event => communityMatch(event, communityFilter))
+    .filter(event => event.title.toLowerCase().includes(query));
+
+  const pageSize = paginationSize === 'All' ? filteredEvents.length : paginationSize;
+
+  const nextPage = () => {
+    if ((currentPage + 1) * pageSize < filteredEvents.length) {
+      setCurrentPage(current => current + 1);
+    }
+  };
+
+  const prevPage = () => {
+    if (currentPage > 0) {
+      setCurrentPage(current => current - 1);
+    }
+  };
+
+  const changePageSize = (e: ChangeEvent<HTMLSelectElement>) => {
+    if (e.target.value === 'All') {
+      setPaginationSize(e.target.value);
+    } else {
+      setPaginationSize(parseInt(e.target.value, 10));
+    }
+  };
+
+  useEffect(() => {
+    setCurrentPage(0);
+  }, [pageSize]);
+
+  const leftPageBound = currentPage * pageSize;
+  const rightPageBound = Math.min((currentPage + 1) * pageSize, filteredEvents.length);
+
   return (
     <div className={style.container}>
-      <h1>Manage Events</h1>
       <div className={style.header}>
-        <div className={style.buttons}>
+        <h1>Manage Events</h1>
+        <div className={style.viewControls}>
+          <IconButton onClick={() => setViewMode('card')}>
+            <BsFillGrid3X3GapFill className={viewMode === 'card' ? style.active : style.inactive} />
+          </IconButton>
+          <IconButton onClick={() => setViewMode('row')}>
+            <FaThList className={viewMode === 'row' ? style.active : style.inactive} />
+          </IconButton>
+        </div>
```

Comment:
It might be nice if it could save the user's preferred view mode in localStorage or in the URL so that it doesn't reset to list view when they edit an event and go back to the list